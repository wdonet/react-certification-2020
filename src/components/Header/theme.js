const base = {
  easeOutBack: 'cubic-bezier(0.34, 1.56, 0.64, 1)',
  colorWhite: 'rgb(255, 255, 255)',
  colorBlack: 'rgb(0, 0, 0)',
};

const dark = {
  id: 'dark',
  ...base,
  backgroundColor: '#333',
  textColor: 'black',
};

const light = {
  id: 'light',
  ...base,
  backgroundColor: '#666',
  textColor: 'white',
};

export const theme = { dark, light };
