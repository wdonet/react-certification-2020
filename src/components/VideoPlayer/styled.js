import styled from 'styled-components';

const VideoIframe = styled.div`
  width: 100%;
  height: 100%;
  min-height: 380px;
`;

export { VideoIframe };
