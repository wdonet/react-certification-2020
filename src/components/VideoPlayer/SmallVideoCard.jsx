import React from 'react';
import styled from 'styled-components';

const StyledSmallVideoCard = styled.div`
  display: flex;
  padding: 4px;
  margin: 4px;
  border-bottom: 1px solid #eaeaea;
  cursor: pointer;
`;

const StyledSmallVideoCardDescription = styled.div`
  height: 100%;
  -webkit-box-flex: 1;
  flex-grow: 1;
  padding: 5px;
  box-sizing: border-box;
`;

const SmallVideoCard = ({ video, onClick }) => {
  const { thumbnails, title, description } = video.snippet;
  return (
    <StyledSmallVideoCard onClick={() => onClick && onClick()}>
      <img src={thumbnails.default.url} alt={title} />
      <StyledSmallVideoCardDescription>{description}</StyledSmallVideoCardDescription>
    </StyledSmallVideoCard>
  );
};

export default SmallVideoCard;
