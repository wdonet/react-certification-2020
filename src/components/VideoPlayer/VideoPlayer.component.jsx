import styled from 'styled-components';
import React from 'react';

const VideoPlayer = ({ className, video }) => {
  return (
    <div className={className}>
      <div dangerouslySetInnerHTML={{ __html: video.player.embedHtml }} />
      <h1>{video.snippet.title}</h1>
      <p>{video.snippet.description}</p>
    </div>
  );
};

const StyledVideoPlayer = styled(VideoPlayer)`
  iframe {
    width: 100%;
    height: 60vh;
  }

  p {
    color: rgba(169, 169, 169, 1);
  }
`;

export default StyledVideoPlayer;
