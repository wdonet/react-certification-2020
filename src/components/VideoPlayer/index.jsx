import React, { useEffect, useRef } from 'react';
import { useYoutubeData } from '../../providers/YoutubeData';

import { VideoIframe } from './styled';

const VideoPlayer = ({ videoId }) => {
  const { iframeAPIReady } = useYoutubeData();
  const player = useRef(null);

  useEffect(() => {
    let done = false;
    if (iframeAPIReady) {
      player.current = new window.YT.Player('player', {
        height: '390',
        width: '640',
        videoId,
        events: {
          onReady: (event) => event.target.playVideo(),
          onStateChange: (event) => {
            if (event.data === window.YT.PlayerState.PLAYING && !done) {
              setTimeout(player.stopVideo, 6000);
              done = true;
            }
          },
        },
      });
    }
  }, [iframeAPIReady]);

  useEffect(() => {
    if (player.current && player.current.loadVideoById)
      player.current.loadVideoById(videoId);
  }, [videoId]);

  return <VideoIframe id="player" />;
};

export default VideoPlayer;
