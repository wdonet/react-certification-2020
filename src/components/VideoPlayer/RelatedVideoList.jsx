import React from 'react';
import styled from 'styled-components';
import SmallVideoCard from './SmallVideoCard';

const StyledVideoList = styled.div`
  overflow: scroll;
  height: calc(100vh - 64px);
  width: 40%;
`;

const RelatedVideoList = ({ videos, playVideoByID }) => {
  return (
    <StyledVideoList id="related-videos-list">
      {videos.map((video) => {
        const key = JSON.stringify(video.id);
        return (
          <div data-testid={`small-caption-${key}`} key={key}>
            <SmallVideoCard
              video={video}
              onClick={() => playVideoByID(video.id.videoId)}
            />
          </div>
        );
      })}
    </StyledVideoList>
  );
};

export default RelatedVideoList;
