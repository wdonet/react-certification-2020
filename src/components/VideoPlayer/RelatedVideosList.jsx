import React, { useContext } from 'react';
import { useHistory } from 'react-router';
import styled from 'styled-components';
import AppContext from '../../providers/AppContext';
import SmallVideoCard from './SmallVideoCard';

const StyledVideoList = styled.div`
  overflow: scroll;
  height: calc(100vh - 64px);
  width: 40%;
`;

const RelatedVideosList = () => {
  const { push } = useHistory();
  const { videosList } = useContext(AppContext);
  return (
    <StyledVideoList id="related-videos-list">
      {videosList.map((video) => {
        const key = JSON.stringify(video.id);
        return (
          <div data-testid={`small-caption-${key}`} key={key}>
            <SmallVideoCard
              video={video}
              onClick={() => push({
                pathname: `/player`,
                search: `?id=${video.id.videoId}`,
                state: video
              }) }
            />
          </div>
        );
      })}
    </StyledVideoList>
  );
};

export default RelatedVideosList;
