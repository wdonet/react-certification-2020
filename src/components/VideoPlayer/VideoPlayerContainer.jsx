import React, { useContext, useEffect } from 'react';
import styled from 'styled-components';
import SearchContext from '../../providers/SearchContext';
import RelatedVideoList from "./RelatedVideoList";

const StyledVideoPlayerContainer = styled.div`
  display: flex;
`;

const VideoPlayerContainer = ({ videoId }) => {

  const { videos } = useContext(SearchContext);
  let player = {};

  useEffect(() => {
    let done = false;
    /* global YT */
    /* eslint no-undef: "error" */
    player = new YT.Player('player', {
      height: 0.5 * Math.max(document.documentElement.clientHeight || 0, window.innerHeight || 0),
      width: '60%',
      videoId,
      events: {
        onReady(event) {
          event.target.playVideo();
        },
        onStateChange(event) {
          if (event.data === YT.PlayerState.PLAYING && !done) {
            setTimeout(player.stopVideo, 6000);
            done = true;
          }
        },
      },
    });
  });

  return (
    <StyledVideoPlayerContainer id="video-player-container">
      <div id="player" />
      <RelatedVideoList videos={videos} playVideoByID={(videoID) => player.loadVideoById(videoID,0)}/>
    </StyledVideoPlayerContainer>
  );
};

export default VideoPlayerContainer;
