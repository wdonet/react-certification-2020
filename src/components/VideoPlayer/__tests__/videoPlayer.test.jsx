import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';

import VideoPlayer from '../index';
import YoutubeDataProvider from '../../../providers/YoutubeData';

import { lightTheme } from '../../../theme';

describe('VideoPlayer', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <YoutubeDataProvider>
          <ThemeProvider theme={lightTheme}>
            <VideoPlayer videoId="999dfdj349" />
          </ThemeProvider>
        </YoutubeDataProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
