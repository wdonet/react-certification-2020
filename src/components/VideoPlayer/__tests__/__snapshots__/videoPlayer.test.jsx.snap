// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer renders correctly 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  min-height: 380px;
}

<div
  className="c0"
  id="player"
/>
`;
